### ISSUE TITLE
OSS Index Analyzer uses the wrong server ID

### ISSUE THREAD

**Issue by akersch** [dependency-check/DependencyCheck#2886]
**Describe the bug**

There seems to be a bug in the OSS Index Analyzer in the Maven plugin. When using a CVE cache with configured cveServerId it seems that the OSS Index Analyzer uses this server ID for calling the OSS Index service. The OSS Index Analyzer then fails. Worse: it seems that the OSS Analyzer sends the username and password of the CVE Analyzer to the OSS Indexing Service. 

**Version of dependency-check used**
The problem occurs using version 6.0.2 of the maven plugin.

**Log file**
When reporting errors, 99% of the time log file output is required. Please post the log file as a [gist](https://gist.github.com/) and provide a link in the new issue.

The log file shows internals, please see the extract of the log below.

**To Reproduce**
1. Configure the maven plugin like:

```
				<plugin>
					<groupId>org.owasp</groupId>
					<artifactId>dependency-check-maven</artifactId>
					<version>${owasp-dependency-check-plugin.version}</version>
					<configuration>
						<cveServerId>myserver</cveServerId>
						<cveUrlModified>https://myserver/my-cache/nvdcve-1.1-modified.json.gz</cveUrlModified>
						<cveUrlBase>https://myserver/my-cache/nvdcve-1.1-%d.json.gz</cveUrlBase>
						<failBuildOnCVSS>7</failBuildOnCVSS>
						<skipProvidedScope>true</skipProvidedScope>
						<connectionTimeout>60000</connectionTimeout>
						<suppressionFiles>
							<file>${basedir}/.dependency-skip.xml</file>
						</suppressionFiles>
						<outputDirectory>${basedir}/target/site</outputDirectory>
						<dataDirectory>${basedir}/target/dependency-check</dataDirectory>
					</configuration>
				</plugin>
```

Configure `myserver` with correct username and password in your Maven `settings.xml`. 

2. Run the dependency-check:

```
mvn clean dependency-check:check
```

The outcome is that, the CVE files are correctly downloaded from the caching server with the correct credentials. But the OSS Index Analyzer then finally fails, it seems to be using the cveServerId credentials:

```
[DEBUG] Adding user/pw from settings.xml as basic authorization
[DEBUG] Connecting to: https://ossindex.sonatype.org/api/v3/component-report
[DEBUG] Begin Analysis of '/Users/kean/.m2/repository/ch/qos/logback/logback-classic/1.2.3/logback-classic-1.2.3.jar' (Sonatype OSS Index Analyzer)
[WARNING] An error occurred while analyzing '/Users/kean/.m2/repository/org/springframework/boot/spring-boot-starter-web/2.3.3.RELEASE/spring-boot-starter-web-2.3.3.RELEASE.jar' (Sonatype OSS Index Analyzer).
[DEBUG] org.owasp.dependencycheck.analyzer.exception.AnalysisException: Invalid credentails provided for OSS Index
    [at org.owasp.depen36dencycheck.analyzer.OssIndexAnalyzer.analyzeDependency  (OssIndexAnalyzer.java:142)
    at org.owasp.dependencycheck.analyzer.AbstractAnalyzer.analyze (AbstractAnalyzer.java:131)
    at org.owasp.dependencycheck.AnalysisTask.call (AnalysisTask.java:88)
    at java.util.concurrent.FutureTsk.run1;36mDEBUG] Begin Analysis of '/Users/kean/.m2/repository /org/glassfish/jakarta.el/3.0.3/jakarta.el-3.0.3.jar' (Sonatype OSS Index Analyzer)
(FutureTask.java:264)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1128)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:628)
    at java.lang.Thread.run (Thread.java:834)
Caused by: org.sonatype.ossindex.service.client.transport.Transport$TransportException[: Unexpected response; status: 401
    at org.sonatype.ossindex.service.client.transport.HttpUrlConnectionTransport.post (HttpUrlConnectionTransport.java:106)
    at org.sonatype.ossindex.service.client.internal.OssindexClientImpl.doRequestComponentReports (OssindexClientImpl.java:204)
    at org.sonatype.ossindex.service.client.internal.OssindexClientImpl.requestComponentReports (OssindexClientImpl.java:170)
    at org.owasp.dependencycheck.analyzer.OssIndexAnalyzer.requestReports (OssIndexAnalyzer.java:198)
    at org.owasp.dependencycheck.analyzer.OssIndexAnalyzer.analyzeDependency (OssIndexAnalyzer.java:138)
    at org.owasp.dependencycheck.analyzer.AbstractAnalyzer.analyze (AbstractAnalyzer.java:131)
    at org.owasp.dependencycheck.AnalysisTask.call (AnalysisTask.java:88)
    at org.owasp.dependencycheck.AnalysisTask.call (AnalysisTask.java:37)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1128)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:628)
    at java.lang.Thread.run (Thread.java:834)

```

When removing the cve settings on the plugin in `pom.xml`, the plugin downloads from the NIST DB and the OSS Index Analyzer succeeds without using basic-auth.

Furthermore, it is not possible to set the ossIndexServerId in the pom.xml, the setting seems to be ignored.

**Expected behavior**

Do not use cveServerId for OSS Index Analysis. User ossIndexServerId for OSS Index Analysis.

**Additional context**

**Comment by akersch** [comment]
Update:

When configuring the ossIndexServerId directly without configuring the cveServerId, the build **correctly** fails, so the ossIndexServerId is taken in this scenario for the OSS Index Analysis:

```
				<plugin>
					<groupId>org.owasp</groupId>
					<artifactId>dependency-check-maven</artifactId>
					<version>${owasp-dependency-check-plugin.version}</version>
					<configuration>
						<ossIndexServerId>myserver</ossIndexServerId>
						<failBuildOnCVSS>7</failBuildOnCVSS>
						<skipProvidedScope>true</skipProvidedScope>
						<connectionTimeout>60000</connectionTimeout>
						<suppressionFiles>
							<file>${basedir}/.dependency-skip.xml</file>
						</suppressionFiles>
						<outputDirectory>${basedir}/target/site</outputDirectory>
						<dataDirectory>${basedir}/target/dependency-check</dataDirectory>
					</configuration>
				</plugin>
```

**Comment by akersch** [comment]
Update:

Trying to unset the ossIndexServerId does not work, the build fails:

```
				<plugin>
					<groupId>org.owasp</groupId>
					<artifactId>dependency-check-maven</artifactId>
					<version>${owasp-dependency-check-plugin.version}</version>
					<configuration>
						<cveServerId>myserver</cveServerId>
						<cveUrlModified>https://myserver/my-cache/nvdcve-1.1-modified.json.gz</cveUrlModified>
						<cveUrlBase>https://myserver/my-cache/nvdcve-1.1-%d.json.gz</cveUrlBase>
						<ossIndexServerId></ossIndexServerId>
						<failBuildOnCVSS>7</failBuildOnCVSS>
						<skipProvidedScope>true</skipProvidedScope>
						<connectionTimeout>60000</connectionTimeout>
						<suppressionFiles>
							<file>${basedir}/.dependency-skip.xml</file>
						</suppressionFiles>
						<outputDirectory>${basedir}/target/site</outputDirectory>
						<dataDirectory>${basedir}/target/dependency-check</dataDirectory>
					</configuration>
				</plugin>
```

**Comment by akersch** [comment]
Update:

Trying to set ossIndexServerId to an invalid server ID does not work, the build fails. Note that the build fails similar to the runs above with invalid credentials and not with server ID not found:

```
				<plugin>
					<groupId>org.owasp</groupId>
					<artifactId>dependency-check-maven</artifactId>
					<version>${owasp-dependency-check-plugin.version}</version>
					<configuration>
						<cveServerId>myserver</cveServerId>
						<cveUrlModified>https://myserver/my-cache/nvdcve-1.1-modified.json.gz</cveUrlModified>
						<cveUrlBase>https://myserver/my-cache/nvdcve-1.1-%d.json.gz</cveUrlBase>
						<ossIndexServerId>invalid</ossIndexServerId>
						<failBuildOnCVSS>7</failBuildOnCVSS>
						<skipProvidedScope>true</skipProvidedScope>
						<connectionTimeout>60000</connectionTimeout>
						<suppressionFiles>
							<file>${basedir}/.dependency-skip.xml</file>
						</suppressionFiles>
						<outputDirectory>${basedir}/target/site</outputDirectory>
						<dataDirectory>${basedir}/target/dependency-check</dataDirectory>
					</configuration>
				</plugin>
```

**Comment by akersch** [comment]
Update:

Trying to disable the OSS Index Analyzer. This does not work, the OSS Index Analyzer is running and fails:

```
				<plugin>
					<groupId>org.owasp</groupId>
					<artifactId>dependency-check-maven</artifactId>
					<version>${owasp-dependency-check-plugin.version}</version>
					<configuration>
						<cveServerId>myserver</cveServerId>
						<cveUrlModified>https://myserver/my-cache/nvdcve-1.1-modified.json.gz</cveUrlModified>
						<cveUrlBase>https://myserver/my-cache/nvdcve-1.1-%d.json.gz</cveUrlBase>
						<ossIndexAnalyzerEnabled>false</ossIndexAnalyzerEnabled>
						<failBuildOnCVSS>7</failBuildOnCVSS>
						<skipProvidedScope>true</skipProvidedScope>
						<connectionTimeout>60000</connectionTimeout>
						<suppressionFiles>
							<file>${basedir}/.dependency-skip.xml</file>
						</suppressionFiles>
						<outputDirectory>${basedir}/target/site</outputDirectory>
						<dataDirectory>${basedir}/target/dependency-check</dataDirectory>
					</configuration>
				</plugin>
```

**Comment by akersch** [comment]
Update:

Disabling the OSS Index Analyzer on the command line, works, the OSS Index Analyzer is not running:

```
mvn clean dependency-check:check -X -DossindexAnalyzerEnabled=false
```

The CVE configuration is as given above.

**Comment by jeremylong** [comment]
A fix will be included in 6.0.3.
